src/main/groovy/com/bmuschko/gradle/docker/tasks/image/Dockerfile.groovy

```diff
@@ -628,56 +936,53 @@ class Dockerfile extends DefaultTask {
     }
 
     static abstract class FileInstruction implements Instruction {
-        private final Object src
-        private final Object dest
-        private final Object flags
+        private final String src
+        private final String dest
+        private final String flags
+        private final Provider<File> provider
```

Comment:
Is this really a `File` or a `Directory`.
You probably want this to be a `RegularFileProperty` or a `DirectoryProperty`:

- https://docs.gradle.org/current/javadoc/org/gradle/api/file/RegularFileProperty.html
- https://docs.gradle.org/current/javadoc/org/gradle/api/file/DirectoryProperty.html